### ISSUE TITLE
PendingMigrationError during local session

### ISSUE THREAD

**Issue by AthulMuralidhar** [thepracticaldev/dev.to#6778]
**Describe the bug**
 -  I've always wanted to contribute to dev.to and what better time to start contributing than this, so I made a fork and started to dev
- I chose the `Docker` method of setting my local instance up because it is most familiar, but ended up with a migration error during loading

**To Reproduce**

- Go to the doc section: https://docs.dev.to/installation/docker/
- In  **Running the Docker app (basic) section**,  I was able to go untill step 2, but I got stuck in step 3. 
This was the particular error:
```bash
Creating network "devto_default" with the default driver
Creating devto_redis_1 ... done
Creating devto_db_1    ... done
rails aborted!
Don't know how to build task 'search:setup' (See the list of available tasks with `rails --tasks`)
```
- being the brave dev that I am, I went on to step 4 and used `docker-compose up` to and jumped into my local server's main page and I got this:

<img width="1253" alt="Screenshot 2020-03-23 at 11 22 06" src="https://user-images.githubusercontent.com/35686547/77306856-89ee3380-6cf8-11ea-98e4-0fad93acbcaa.png">

**Expected behavior**

- I suspect that the yml file for docker is a bit messy, as I have some experience here. This is the first time that I am setting my local instance up

**Desktop (please complete the following information):**

- OS: macOS Catalina v10.15.3
- Browser: Google Chrome (tested this behaviour in both normal and incognito mode)
- Version:  v80.0.3987.149 (Official Build) (64-bit)

Same behaviour on:
- OS: Ubuntu 19.10
- Browser: Firefox v74.0 (64-bit)

#### Note:
- Also please feel free to close this if irrelevant :)

**Comment by rhymes** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by rhymes** [comment]
Hi @AthulMuralidhar, I tried to replicate the error but for me it failed here:

```shell
$ docker-compose build
...
...
[1/5] Validating package.json...
error dev.to@1.0.0: The engine "yarn" is incompatible with this module. Expected version "1.21.x". Got "1.22.1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
ERROR: Service 'webpacker' failed to build: The command '/bin/sh -c yarn install && yarn check --integrity' returned a non-zero code: 1
```

:-(

**Comment by AthulMuralidhar** [comment]
can i claim this issue? @rhymes

**Comment by AthulMuralidhar** [comment]
> Hi @AthulMuralidhar, I tried to replicate the error but for me it failed here:
> 
> ```shell
> $ docker-compose build
> ...
> ...
> [1/5] Validating package.json...
> error dev.to@1.0.0: The engine "yarn" is incompatible with this module. Expected version "1.21.x". Got "1.22.1"
> error Found incompatible module.
> info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
> ERROR: Service 'webpacker' failed to build: The command '/bin/sh -c yarn install && yarn check --integrity' returned a non-zero code: 1
> ```

maybe check what's up with the yarn package in your local docker instance @rhymes?

**Comment by citizen428** [comment]
@AthulMuralidhar Can you please try to run `docker-compose run web rails db:migrate` after `docker-compose run web rails db:setup`?

**Comment by citizen428** [comment]
@AthulMuralidhar I see two issues here:

1. The pending migrations. Please try to run `docker-compose run web rails db:migrate`.
2. The error about the `search:setup` Rake task not being found. This should not happen if you're on a recent version of the code.

On the other hand I'm seeing the same issue as @rhymes right now, this seems to point towards a possible solution:

https://github.com/rails/webpacker/issues/752

So @AthulMuralidhar, if you want to work on this, I'd start off the assumption that Docker is working (maybe we need to add the migrate step to the docs though) and that there are things around it that we may need to address first.

**Comment by citizen428** [comment]
The yarn error actually has to do with our `package.json`: `"yarn": "1.21.x"`

I'm currently working on this, so will assign myself the issue @AthulMuralidhar. I'll also try to address the other points you raised.

**Comment by AthulMuralidhar** [comment]
> @AthulMuralidhar I see two issues here:
> 
> 1. The pending migrations. Please try to run `docker-compose run web rails db:migrate`.
> 2. The error about the `search:setup` Rake task not being found. This should not happen if you're on a recent version of the code.
> 
> On the other hand I'm seeing the same issue as @rhymes right now, this seems to point towards a possible solution:
> 
> [rails/webpacker#752](https://github.com/rails/webpacker/issues/752)
> 
> So @AthulMuralidhar, if you want to work on this, I'd start off the assumption that Docker is working (maybe we need to add the migrate step to the docs though) and that there are things around it that we may need to address first.

- 1) seems to be a solution for the migration thing
- 2) I would need to investigate it a bit further, let me check the versions a bit

**Comment by AthulMuralidhar** [comment]
i'll  keep things posted here :) and good luck @citizen428

**Comment by citizen428** [comment]
Great, thanks for the feeback so far 👍 Re 1 I'll probably just add a line to the docs. The second thing really shouldn't happen unless you have a pretty old checkout and didn't recently pull.

**Comment by AthulMuralidhar** [comment]
@citizen428 the migration issue is gone :tada but I have some other issue with the `search:setup` part

**Comment by citizen428** [comment]
@AthulMuralidhar We are working on said channel, in the meantime feel free to open a new issue. I'll close this one now, as it's original problem has been resolved.
